web/src/components/FormationDateModal.tsx

```diff
@@ -43,7 +50,7 @@ export const FormationDateModal = (props: Props): ReactElement => {
       onValidation("dateOfFormation", true);
       return;
     }
-    if (!profileData.municipality || fieldStates.municipality.invalid) {
+    if (shouldShowMunicipalityQuestion() && (!profileData.municipality || fieldStates.municipality.invalid)) {
```

Comment:
I think this logic should be combined. If `shouldShowMunicipalityQuestion` determines (as the name states) whether the municipality question is shown, I'd expect all that logic to live in the function. Here, it's only handling *some* of the logic around when-to-show, and the rest is in the if-clause. That logic should be moved into the function too.